resnet/main.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+"""Example script to train a ResNet model on ImageNet."""
+
+import os
+import sys
+
+import torch
+from composer import Trainer
+from composer.algorithms import (EMA, SAM, BlurPool, ChannelsLast, ColOut,
+                                 LabelSmoothing, MixUp, ProgressiveResizing,
+                                 RandAugment, StochasticDepth)
+from composer.callbacks import MemoryMonitor, LRMonitor, SpeedMonitor
+from composer.loggers import ProgressBarLogger, WandBLogger
+from composer.optim import CosineAnnealingWithWarmupScheduler, DecoupledSGDW
+from composer.utils import dist
+from omegaconf import OmegaConf
+
+from data import build_imagenet_dataspec, build_streaming_imagenet_dataspec
+from model import build_composer_resnet
+
+
+def build_logger(name, kwargs):
```

Comment:
type annotations for function params? (also applies to other helper functions). Config is an omegaconf.DictConfig

Reply:
I think I added the right ones?